Am/eevee

anniemated opened the pull request:
This is my first ever coding project so it's a little scuffed, but it's a slightly modified version of the game on the hackapet website! You play as Eevee, and you dodge chocolate (poisonous to dogs and foxes) while collecting biscuits (Electric, Water, and Fire type for Jolteon, Vaporeon, and Flareon)!

tanmayrajk commented:
The game looks very cute! I noticed that the character can go out of the screen, you should limit the movement of the character inside the screen. In your code where you've did the movement, just add a check to see if the character's near the edge of the screen and if so don't change its position.

tanmayrajk commented:
That's all you need to do!
```
    if game_over == False:
        if keys[pygame.K_LEFT] and eevee_sprite.x > 0:
            eevee_sprite.x -= speed
        if keys[pygame.K_RIGHT] and eevee_sprite.x < 98:
            eevee_sprite.x += speed
```
![python_JJl3pzFEkK](https://github.com/user-attachments/assets/ffed6a74-3b73-4895-9347-438b6b9e1580)

acornitum commented:
This game is _adorable_!! Honestly some of the best art I've seen in Hackapet submissions - I'm also a huge pokemon fan. I love the small details from the swinging tail on the clock, to the bobbing on the eevee! The background is honestly so cute too.

However, there are some things I'd like you to add before I merge it! 
- A start screen! Especially since you've got the art skills to make something look really nice - I'd really love it if you could add one.
- Could there be a score counter of some sort? _Alternatively_, I think it would be really cool that, when you collect enough of one of the cookies, the eevee evolves to the eeveelution that the cookie is of! Either way, some sort of positive feedback for the player would take this game up a notch.

I think this is still really good for your first time coding - you've made a functional game - and you should be proud of yourself! I can't wait to see what the game looks like after the feedback has been implemented.

_Note: don't open a new PR, just commit changes to this current thread + send a comment when changes have been made!_

anniemated commented:
Hi Acon, I added the start screen and fixed the going out of the screen thing that Tan recommended, but I tried really hard to make the Eeveelution thing work and it just isn't working... I've spent like way too much time on something that should be pretty simple but my code is so spaghetti that I have no idea what's causing these random issues (in one of my branches it only turns into Vaporeon and Flareon, in another one there's like a spectral Eevee left behind by the player, etc). I asked my sibling who's actually good at coding and he said that it would be more productive to rewrite it from the ground up 😭 😭 😭 

Because of the new semester my workload has increased a lot and I have a bunch of outside of school commitments now, and I don't know if I can keep working on this (or if it's even worth it to try and fix it). Could you please just let this game slide through? I really am proud of the art and effort I put into this! Working on Hackapet was really fun and I want something to remember my first project by 😢

acornitum commented:
Yup! I love the start screen and I appreciate the effort for the evolution parts of it :D 

Going to merge this - fill out the form here: https://forms.hackclub.com/t/isDdJtrnrvus

Thanks so much for submitting!
